### ISSUE TITLE
revert to original type of combo boxes for screen reader accessibility

### ISSUE THREAD

**Issue by RoryMichie** [gradio-app/gradio#6463]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
As gradio has been updated and a new type of combo box was developed using custom javascript, the screen reader accessibility has lessened; there are two reasons for this:
(1) The new combo boxes do not appear to screen readers as combo boxes, and thus cannot be navigated to using traditional navigation keystrokes
(2) The most recent implementation poses significant problems for screen readers, because they are unable to find or select elements in the combo boxes without solutions which are significantly hackier than is probably reasonable
**Describe the solution you'd like**  
I would like there to be a feature when launching the interface that allows combo boxes to be reverted back to their original implementation.
This is to say that they will once again be created with <select>, and not using the custom implementation

**Additional context**  
I have tested with [nvda](https://nvaccess.org) but have not done so with any others. The issues will probably carry over to others, however I can test with jaws for windows and voice over for ios if these tests would help.
Thanks for reading.

**Comment by DJ-Wolfy** [comment]
Sorry, it seems like there was a problem posting this. When I say using the original implementation  instead of the custom implementation, I mean to use <select> (less than select greater than)

**Comment by abidlabs** [comment]
Hi @DJ-Wolfy by combo boxes, do you mean `gr.Dropdown()`?

**Comment by DJ-Wolfy** [comment]
Hi,
Yes, gr.Dropdown sorry.

**Comment by abidlabs** [comment]
Thanks for clarifying. We are unlikely to revert to native select elements because they come with many disadvantages including lack of support for custom values and search. However, we do want to ensure that the `gr.Dropdown` component is accessible to screen readers. Just to confirm, the issue right now is that the `gr.Dropdown` is not navigable by keyboard?

**Comment by DJ-Wolfy** [comment]
Yes. However, the implementation before the current one, which used buttons, was great, and worked well. I was thinking some sort of way to switch between them

**Comment by DJ-Wolfy** [comment]
Update with some extra info:
Things may not be as bad as initially thought, at least in part -- selection of an item by entering enough information that there is no ambiguity and then pressing the enter key does still select the item.
Could gr.SimpleDropdown be what I want? I can't seem to use it directly. Hmm...

**Comment by abidlabs** [comment]
Can you elaborate what behavior is currently not possible with a screen reader? cc @hannahblair

**Comment by DJ-Wolfy** [comment]
Hi,
Thanks for the response.
The expected behaviour is either:
A) The item appears as a combo box, and you can use the arrows to navigate through items.
Or,
B) The item appears as an edit box, which, when focused, has buttons below that dynamically show and hide based on items which currently match the search string
The current behaviour is inconsistent from machine to machine and session to session, however:
Always, the edit box appears, and when focused, a list appears below. Keyboard navigation doesn't work in the list -- the object tree must be navigated when entering it.
Using nvda+enter to *activate* the item sometimes selects it, but it sometimes doesn't, and I haven't been able to figure out exactly what causes this to happen as it seems to work and not work on the same code from session to session and computer to computer.
Hope this helps. Let me know if any more information is required. I will make a recording later to demonstrate the behaviour in gradio with the HTML select combo boxes, gradio with the combo box with button implementation, and gradio with the current implementation if this would also be helpful.

**Comment by abidlabs** [comment]
Thanks for raising this @RoryMichie, and apologies for the slow follow-up. The Dropdown has been reworked a fair bit since this was filed. We're not going to revert to native <select>, but accessibility is something we do care about. If the current implementation still has screen reader issues on a recent version, could you open a new issue with the specific keystrokes/behavior that break? That'll be much more actionable. Closing this one as stale in the meantime.
